### PR TITLE
Add methods median, ward.d, ward.d2 and mcquitty

### PR DESCRIPTION
**Pull request by koenderks**
Implement feature request https://github.com/jasp-stats/jasp-issues/issues/921

For hierarchical clustering, add methods `ward.D`, `ward.D2`, `mcquitty`, and `median`.

### CODE REVIEW

File: Resources/Machine Learning/qml/mlClusteringHierarchical.qml [hunk 1]
```diff
@@ -159,7 +159,11 @@ Form {
...
 					{ label: qsTr("Single"), value: "single"},
 					{ label: qsTr("Complete"), value: "complete"},
-					{ label: qsTr("Centroid"), value: "centroid"}
+					{ label: qsTr("Centroid"), value: "centroid"},
+                    { label: qsTr("Median"), value: "median"},
+                    { label: qsTr("Ward.D"), value: "ward.D"},
+                    { label: qsTr("Ward.D2"), value: "ward.D2"},
+                    { label: qsTr("McQuitty"), value: "mcquitty"}
                 ]
             }
```
Comment (vandenman) [inline Resources/Machine Learning/qml/mlClusteringHierarchical.qml]:
```suggestion
					{ label: qsTr("Median"), value: "median"},
					{ label: qsTr("Ward.D"), value: "ward.D"},
					{ label: qsTr("Ward.D2"), value: "ward.D2"},
					{ label: qsTr("McQuitty"), value: "mcquitty"}
				]
			}
```
(tabs, not spaces)

**Comment by AlexanderLyNL** [comment]
I only see a change in the qml. Did anything happen other than an argument being passed to the R function? Or do you still need to do something on the R side?

**Comment by koenderks** [comment]
@AlexanderLyNL No this is it

**Comment by AlexanderLyNL** [comment]
It's great that it's so easy. Can you please add unit tests?

**Comment by AlexanderLyNL** [comment]
In principe I can merge this, but the audit unit tests failed.

**Comment by koenderks** [comment]
I think you can merge this now @AlexanderLyNL

**Pull request merged**